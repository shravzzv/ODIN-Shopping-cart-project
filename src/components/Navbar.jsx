import { Link, useLocation } from 'react-router-dom'
import '../styles/components/Navbar.css'
import { useEffect, useState } from 'react'
import PropTypes from 'prop-types'

Navbar.propTypes = {
  cartItemsCount: PropTypes.number.isRequired,
}

export default function Navbar({ cartItemsCount }) {
  const [activeLink, setActiveLink] = useState(1)
  const [isFloating, setIsFloating] = useState(false)
  const { pathname } = useLocation()

  useEffect(() => {
    const handleScroll = () => {
      window.scrollY == 0 ? setIsFloating(false) : setIsFloating(true)
    }
    handleScroll()
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  useEffect(() => {
    if (pathname == '/') setActiveLink(1)
    if (pathname == '/shop') setActiveLink(2)
    if (pathname == '/cart') setActiveLink(3)
    if (!['/', '/shop', '/cart'].includes(pathname)) {
      setActiveLink(null)
    }
  }, [pathname])

  return (
    <nav className={isFloating ? 'floating' : ''}>
      <ul>
        <li>
          <Link
            to='/'
            className={activeLink == 1 ? 'active' : ''}
            onClick={() => setActiveLink(1)}
          >
            <svg
              xmlns='http://www.w3.org/2000/svg'
              height='24'
              viewBox='0 -960 960 960'
              width='24'
            >
              <path d='M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z' />
            </svg>
            <span>Home</span>
          </Link>
        </li>
        <li>
          <Link
            to='/shop'
            className={activeLink == 2 ? 'active' : ''}
            onClick={() => setActiveLink(2)}
          >
            {activeLink == 2 ? (
              <svg
                xmlns='http://www.w3.org/2000/svg'
                height='24'
                viewBox='0 -960 960 960'
                width='24'
              >
                <path d='M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13Z' />
              </svg>
            ) : (
              <svg
                xmlns='http://www.w3.org/2000/svg'
                height='24'
                viewBox='0 -960 960 960'
                width='24'
              >
                <path d='M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z' />
              </svg>
            )}
            <span>Shop</span>
          </Link>
        </li>
        <li>
          <Link
            to='/cart'
            className={activeLink == 3 ? 'active' : ''}
            onClick={() => setActiveLink(3)}
          >
            {activeLink == 3 ? (
              <svg
                xmlns='http://www.w3.org/2000/svg'
                height='24'
                viewBox='0 -960 960 960'
                width='24'
              >
                <path d='M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM208-800h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Z' />
              </svg>
            ) : (
              <svg
                xmlns='http://www.w3.org/2000/svg'
                height='24'
                viewBox='0 -960 960 960'
                width='24'
              >
                <path d='M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z' />
              </svg>
            )}
            <span>Cart</span>
            {cartItemsCount > 0 && (
              <div className='badgeContainer' data-testid='badgeContainer'>
                <span className='badgeLabel' data-testid='badgeLabel'>
                  {cartItemsCount >= 1000 ? '999+' : cartItemsCount}
                </span>
              </div>
            )}
          </Link>
        </li>
      </ul>
    </nav>
  )
}
