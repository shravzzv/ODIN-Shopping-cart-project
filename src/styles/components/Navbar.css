nav {
  background-color: var(--surface);
  color: var(--primary);
  padding: 10px 0;

  &.floating {
    background-color: var(--surface-container);
    padding: 0;
    border-radius: 9999px;
    position: fixed;
    left: 50%;
    transform: translate(-50%);
  }

  ul {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  a {
    color: var(--primary);
    text-decoration: inherit;
    display: flex;
    align-items: center;
    justify-content: center;
    font-weight: 500;
    border-radius: 9999px;
    padding: 10px 20px;
    transition: background-color 300ms ease-in-out;
    border: 2px solid transparent;

    & svg {
      fill: var(--primary);
    }

    .badgeContainer {
      background-color: var(--error);
      border-radius: 9999px;
      padding: 4px;

      span {
        color: var(--on-error);
        font-size: 12px;
        display: flex;
        align-items: center;
        justify-content: center;
      }
    }

    &.active {
      background-color: var(--secondary-container);
    }
  }
}

@media (max-width: 767px) {
  /* mobile exclusive */
  nav {
    a {
      flex-direction: column;
      position: relative;

      > span {
        font-size: 0.75rem;
      }

      .badgeContainer {
        position: absolute;
        top: 0;
        left: 50%;
      }
    }

    &.floating {
      bottom: 20px;
      -webkit-animation: slide-in-bottom 0.5s
        cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
      animation: slide-in-bottom 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
    }
  }
}

@media (min-width: 768px) {
  nav {
    a {
      padding: 20px 30px;
      gap: 5px;
    }

    &.floating {
      top: 10px;
      -webkit-animation: slide-in-top 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94)
        both;
      animation: slide-in-top 0.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) both;
    }
  }
}

@media (min-width: 1024px) {
  nav {
    a {
      &:hover {
        border-color: var(--primary);
      }
    }
  }
}
