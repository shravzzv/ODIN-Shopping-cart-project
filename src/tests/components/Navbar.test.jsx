import { describe, test, expect } from 'vitest'
import Navbar from '../../components/Navbar'
import { screen, render, fireEvent } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { MemoryRouter } from 'react-router-dom'

describe('Navbar', () => {
  describe('should have the correct structure', () => {
    test('should show three links', () => {
      render(
        <MemoryRouter>
          <Navbar cartItemsCount={1} />
        </MemoryRouter>
      )
      expect(screen.getAllByRole('link')).toHaveLength(3)
    })

    test('the three links should be for home, shop and cart', () => {
      render(
        <MemoryRouter>
          <Navbar cartItemsCount={1} />
        </MemoryRouter>
      )

      const homeLink = screen.getByRole('link', { name: /home/i })
      const shopLink = screen.getByRole('link', { name: /shop/i })
      const cartLink = screen.getByRole('link', { name: /cart/i })

      expect(homeLink).toBeInTheDocument()
      expect(shopLink).toBeInTheDocument()
      expect(cartLink).toBeInTheDocument()
      expect(homeLink).toHaveAttribute('href', '/')
      expect(shopLink).toHaveAttribute('href', '/shop')
      expect(cartLink).toHaveAttribute('href', '/cart')
    })
  })

  describe('should have the correct functionality', () => {
    test('should show the active link when clicked', async () => {
      const user = userEvent.setup()

      const { container } = render(
        <MemoryRouter>
          <Navbar cartItemsCount={1} />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()

      const homeLink = screen.getByRole('link', { name: /home/i })
      const shopLink = screen.getByRole('link', { name: /shop/i })
      const cartLink = screen.getByRole('link', { name: /cart/i })

      expect(homeLink).toHaveClass('active')
      expect(shopLink).not.toHaveClass('active')
      expect(cartLink).not.toHaveClass('active')

      await user.click(shopLink)
      expect(homeLink).not.toHaveClass('active')
      expect(shopLink).toHaveClass('active')
      expect(cartLink).not.toHaveClass('active')

      await user.click(cartLink)
      expect(homeLink).not.toHaveClass('active')
      expect(shopLink).not.toHaveClass('active')
      expect(cartLink).toHaveClass('active')

      await user.click(homeLink)
      expect(homeLink).toHaveClass('active')
      expect(shopLink).not.toHaveClass('active')
      expect(cartLink).not.toHaveClass('active')
    })

    test('should be floating when the window is scrolled', async () => {
      const { container } = render(
        <MemoryRouter>
          <Navbar cartItemsCount={1} />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()

      const nav = screen.getByRole('navigation')
      expect(nav).not.toHaveClass('floating')

      // Simulate a scroll event using fireEvent.scroll
      await fireEvent.scroll(window, { target: { scrollY: 300 } })
      expect(nav).toHaveClass('floating')

      // Should not be floating when the window is not scrolled
      await fireEvent.scroll(window, { target: { scrollY: 0 } })
      expect(nav).not.toHaveClass('floating')
    })

    test('should not show cart items count if it is zero', () => {
      const { container } = render(
        <MemoryRouter>
          <Navbar cartItemsCount={0} />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()

      expect(screen.queryByTestId('badgeContainer')).not.toBeInTheDocument()
      expect(screen.queryByTestId('badgeLabel')).not.toBeInTheDocument()
    })

    test('should show cart items count if it is greater than 0', () => {
      const { container } = render(
        <MemoryRouter>
          <Navbar cartItemsCount={1} />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()

      expect(screen.queryByTestId('badgeContainer')).toBeInTheDocument()
      const labelCount = screen.queryByTestId('badgeLabel')
      expect(labelCount).toBeInTheDocument()
      expect(labelCount.textContent).toMatch('1')
    })

    test('should show cart items count as 999+ if it is >= 1000', () => {
      const { container } = render(
        <MemoryRouter>
          <Navbar cartItemsCount={1000} />
        </MemoryRouter>
      )

      expect(container).toMatchSnapshot()

      expect(screen.queryByTestId('badgeContainer')).toBeInTheDocument()
      const labelCount = screen.queryByTestId('badgeLabel')
      expect(labelCount).toBeInTheDocument()
      expect(labelCount.textContent).toMatch('999+')
    })
  })
})
