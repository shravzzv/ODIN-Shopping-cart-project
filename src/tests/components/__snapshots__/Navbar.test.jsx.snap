// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should have the correct functionality > should be floating when the window is scrolled 1`] = `
<div>
  <nav
    class=""
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z"
            />
          </svg>
          <span>
            Shop
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
            />
          </svg>
          <span>
            Cart
          </span>
          <div
            class="badgeContainer"
            data-testid="badgeContainer"
          >
            <span
              class="badgeLabel"
              data-testid="badgeLabel"
            >
              1
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar > should have the correct functionality > should not show cart items count if it is zero 1`] = `
<div>
  <nav
    class=""
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z"
            />
          </svg>
          <span>
            Shop
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
            />
          </svg>
          <span>
            Cart
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar > should have the correct functionality > should show cart items count as 999+ if it is >= 1000 1`] = `
<div>
  <nav
    class=""
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z"
            />
          </svg>
          <span>
            Shop
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
            />
          </svg>
          <span>
            Cart
          </span>
          <div
            class="badgeContainer"
            data-testid="badgeContainer"
          >
            <span
              class="badgeLabel"
              data-testid="badgeLabel"
            >
              999+
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar > should have the correct functionality > should show cart items count if it is greater than 0 1`] = `
<div>
  <nav
    class=""
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z"
            />
          </svg>
          <span>
            Shop
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
            />
          </svg>
          <span>
            Cart
          </span>
          <div
            class="badgeContainer"
            data-testid="badgeContainer"
          >
            <span
              class="badgeLabel"
              data-testid="badgeLabel"
            >
              1
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navbar > should have the correct functionality > should show the active link when clicked 1`] = `
<div>
  <nav
    class=""
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M180-120q-25 0-42.5-17.5T120-180v-76l160-142v278H180Zm140 0v-160h320v160H320Zm360 0v-328L509-600l121-107 190 169q10 9 15 20.5t5 24.5v313q0 25-17.5 42.5T780-120H680ZM120-310v-183q0-13 5-25t15-20l300-266q8-8 18.5-11.5T480-819q11 0 21.5 3.5T520-804l80 71-480 423Z"
            />
          </svg>
          <span>
            Home
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/shop"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M841-518v318q0 33-23.5 56.5T761-120H201q-33 0-56.5-23.5T121-200v-318q-23-21-35.5-54t-.5-72l42-136q8-26 28.5-43t47.5-17h556q27 0 47 16.5t29 43.5l42 136q12 39-.5 71T841-518Zm-272-42q27 0 41-18.5t11-41.5l-22-140h-78v148q0 21 14 36.5t34 15.5Zm-180 0q23 0 37.5-15.5T441-612v-148h-78l-22 140q-4 24 10.5 42t37.5 18Zm-178 0q18 0 31.5-13t16.5-33l22-154h-78l-40 134q-6 20 6.5 43t41.5 23Zm540 0q29 0 42-23t6-43l-42-134h-76l22 154q3 20 16.5 33t31.5 13ZM201-200h560v-282q-5 2-6.5 2H751q-27 0-47.5-9T663-518q-18 18-41 28t-49 10q-27 0-50.5-10T481-518q-17 18-39.5 28T393-480q-29 0-52.5-10T299-518q-21 21-41.5 29.5T211-480h-4.5q-2.5 0-5.5-2v282Zm560 0H201h560Z"
            />
          </svg>
          <span>
            Shop
          </span>
        </a>
      </li>
      <li>
        <a
          class=""
          href="/cart"
        >
          <svg
            height="24"
            viewBox="0 -960 960 960"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280-80q-33 0-56.5-23.5T200-160q0-33 23.5-56.5T280-240q33 0 56.5 23.5T360-160q0 33-23.5 56.5T280-80Zm400 0q-33 0-56.5-23.5T600-160q0-33 23.5-56.5T680-240q33 0 56.5 23.5T760-160q0 33-23.5 56.5T680-80ZM246-720l96 200h280l110-200H246Zm-38-80h590q23 0 35 20.5t1 41.5L692-482q-11 20-29.5 31T622-440H324l-44 80h480v80H280q-45 0-68-39.5t-2-78.5l54-98-144-304H40v-80h130l38 80Zm134 280h280-280Z"
            />
          </svg>
          <span>
            Cart
          </span>
          <div
            class="badgeContainer"
            data-testid="badgeContainer"
          >
            <span
              class="badgeLabel"
              data-testid="badgeLabel"
            >
              1
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
